providers/google.go

```diff
@@ -178,8 +186,16 @@ func (p *GoogleProvider) Redeem(ctx context.Context, redirectURL, code string) (
 // account credentials.
 func (p *GoogleProvider) SetGroupRestriction(groups []string, adminEmail string, credentialsReader io.Reader) {
 	adminService := getAdminService(adminEmail, credentialsReader)
-	p.GroupValidator = func(email string) bool {
-		return userInGroup(adminService, groups, email)
+	p.GroupValidator = func(s *sessions.SessionState) bool {
+		// Reset our saved Groups in case membership changed
+		// This is used by `Authorize` on every request
+		s.Groups = make([]string, 0, len(groups))
+		for _, group := range groups {
+			if userInGroup(adminService, group, s.Email) {
+				s.Groups = append(s.Groups, group)
+			}
+		}
+		return len(s.Groups) > 0
```

Comment:
Just to confirm, this is what populates the groups for the session? I think this can be improved though maybe in a later PR.

Would it not be simpler to just call the groups API as is done in Dex (https://github.com/dexidp/dex/pull/1185/files) and list all groups and pop these into the session groups? I wonder why this wasn't done in the first place?

Reply:
Yeah - at this stage I'm just refactoring the existing calls to check membership in groups that are explicitly set in the google groups config options and adding those to the refactored `session.Groups`

Agreed - another refactor PR we should just grab all groups and toss them in the session.